File: javascript/hello-world/hello-world.js
"use strict";
var HelloWorld = (function () {
    function HelloWorld() {
    }
    HelloWorld.prototype.hello = function (name) {
        if (name) {
            return "Hello, " + name + "!";
        }
        else {
            return "Hello, World!";
        }
    };
    return HelloWorld;
}());
exports.HelloWorld = HelloWorld;

File: go/hello-world/hello_world.go
package hello

const testVersion = 2

// HelloWorld greets the user.
// If the user does not supply an argument, it returns "Hello, World!"
// If the user supplies an argument, it returns "Hello, <arg>!"
func HelloWorld(name string) string {
	if name == "" {
		return "Hello, World!"
	}

	return "Hello, " + name + "!"
}
